docs/badges/distribution.rst

```diff
@@ -0,0 +1,311 @@
+
+Credly badge templates can be created manually, but more preferable way is to use bulk operations.
+From ``CredlyOrganization`` list:
+
+- ``sync_organization_badge_templates`` admin list view action
+- ``sync_organization_badge_templates`` management command
+
+Both options perform the same:
+
+1. Pull available for Organization badge templates from `Credly API`_:
+
+.. code-block::
+
+    GET /organizations/<organization_id>/badge_templates
+
+2. Create ``CredlyBadgeTemplate`` records for each available template.
+
+Badge templates synchronization
+###############################
+
+After the initial Credly badge templates creation on Open edX Credentials side, system starts listen to their updates via available web hooks:
+
+- ``badge_template.created`` (event is triggered whenever a badge template is created)
+    - system creates new ``CredlyBadgeTemplate`` in inactive state;
+    - badges are not start being processed until manual activation;
+- ``badge_template.changed`` (event is triggered whenever a badge template is changed)
+    - system updates corresponding ``CredlyBadgeTemplate`` relevant attributes;
+    - badges processing continues;
+- ``badge_template.deleted`` (event is triggered whenever a badge template is delete)
+    - system archives corresponding ``CredlyBadgeTemplate``;
+    - badges processing stops;
+
+Requirements setup
+~~~~~~~~~~~~~~~~~~
+
+Created badges `must be configured`_.
+
+Desired Requirements are created for each badge template manually:
+
+- create a Requirement for a single course with specific ID for completion event.
+- create a Requirement for a single course with specific ID for revocation event (reverse effect).
+
+CCX courses
+###########
+
+    Both event types (for passing course grade and failing course grade) carry additional information for custom courses identification.
+
+Awarded badges
+~~~~~~~~~~~~~~
+
+Already awarded badges are stored as `UserCredential`_ records.
+
+Credly backend manages its own extended version of user credential:
+
+.. code-block::
+
+    CredlyBadge(UserCredential):
+        """
+        Earned badge.
+        """
+        - - - unified attrs - - -
+        ...
+        - uuid
+        - username: <username>
+        - status: "awarded" | "revoked"
+        ...
+
+        - - - specific attrs - - -
+
+        - issued_badge_id: <uuid>
+        - issued_at: <timestamp>
+        - accept_badge_url: <URL>
+
+
+Side effects
+~~~~~~~~~~~~
+
+`Badge Collector`_ emits signals when:
+
+- a badge is `awarded`_
+- a badge is `revoked`_
+
+There are events which may be used as triggers for distribution **side effects**:
+
+- on award:
+    - credentials.apps.badges.distribution.credly.handlers.notify_user_awarding
+    - credentials.apps.badges.distribution.credly.handlers.issue_badge
+- on revoke:
+    - credentials.apps.badges.distribution.credly.handlers.notify_user_revocation
+    - credentials.apps.badges.distribution.credly.handlers.revoke_badge
```

Comment:
There are times when partners of ours don't want to message learners' when credentials are awarded, mostly because the system is designed in a way to award Credentials when they earn a passing grade (but may have not finished the course). 

Will these notifications be optional?

Reply:
User notifications are still being discussed.
In this docs, it is mostly a hook for a discussion. An option to think about.
We update the docs based on decisions made.